### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by JieningTao**

### CODE REVIEW

File: Assets/Scripts/CameraFollowScript.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraFollowScript : MonoBehaviour {
```
Comment (DavidJB526) [inline Assets/Scripts/CameraFollowScript.cs]: Move curly bracket to next line

File: Assets/Scripts/CameraFollowScript.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        offest = transform.position - Player.transform.position;
+    }
+	
+	// Update is called once per frame
+	void Update ()
+    {
+        transform.position = new Vector3( Player.transform.position.x+offest.x, Player.transform.position.y+offest.y,viewdistance*-1);
+        
+    }
+}
```
Comment (DavidJB526) [inline Assets/Scripts/CameraFollowScript.cs:23]: Consider using Cinemachine instead

File: Assets/Scripts/CameraFollowScript.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraFollowScript : MonoBehaviour {
+
+    [SerializeField]
+    private GameObject Player;
+    [SerializeField]
+    private int viewdistance;
+
```
Comment (DavidJB526) [inline Assets/Scripts/CameraFollowScript.cs]: Remove whitespace

File: Assets/Scripts/CameraFollowScript.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    void Start ()
+    {
+        offest = transform.position - Player.transform.position;
+    }
+	
+	// Update is called once per frame
+	void Update ()
+    {
+        transform.position = new Vector3( Player.transform.position.x+offest.x, Player.transform.position.y+offest.y,viewdistance*-1);
+
```
Comment (DavidJB526) [inline Assets/Scripts/CameraFollowScript.cs:21]: Remove whitespace

File: Assets/Scripts/Checkpoint.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Checkpoint : MonoBehaviour
+{
+    [SerializeField]
+    private Color inactiveColor, activeColor;
+
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs]: Remove whitespace

File: Assets/Scripts/Checkpoint.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    private SpriteRenderer spriteRenderer;
+    private Vector3 shape;
+
+    private void Start()
+    {
+        spriteRenderer = GetComponent<SpriteRenderer>();
+        shape = transform.localScale;
+        audioSource = GetComponent<AudioSource>();
+    }
+    private void UpdateColor()
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs:23]: Add whitespace

File: Assets/Scripts/Checkpoint.cs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        Color color = inactiveColor;
+        if (isactivated)
+            color = activeColor;
+
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs]: Remove whitespace

File: Assets/Scripts/Checkpoint.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        spriteRenderer.color = color;
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs]: Remove whitespace

File: Assets/Scripts/Checkpoint.cs [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+            if (collision.gameObject.CompareTag("Player") && !isactivated)
+            {
+                Debug.Log("Check Point Reached!");
+                PlayerMoveScript player = collision.GetComponent<PlayerMoveScript>();
+                player.setcurrentcheckpoint(this);
+            
+                audioSource.Play();
+            }
+
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs]: Remove whitespace

File: Assets/Scripts/Checkpoint.cs [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+                Debug.Log("Check Point Reached!");
+                PlayerMoveScript player = collision.GetComponent<PlayerMoveScript>();
+                player.setcurrentcheckpoint(this);
+            
+                audioSource.Play();
+            }
+        
+    }
+
+    public void setisactivated(bool value)
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs]: Capitalize Function

File: Assets/Scripts/Door.cs [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class Door : MonoBehaviour
+{
+    [SerializeField]
+    private string sceneToLoad;
+    private bool isPlayerInTrigger;
+
+    //private void OnTriggerStay2D(Collider2D collision)
```
Comment (DavidJB526) [inline Assets/Scripts/Door.cs]: Remove these if you're not using them

File: Assets/Scripts/Door.cs [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    //        }
+    //    }
+    //}
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.CompareTag("Player"))
+            isPlayerInTrigger = true;
+    }
+    private void OnTriggerExit2D(Collider2D collision)
```
Comment (DavidJB526) [inline Assets/Scripts/Door.cs:19]: Add whitespace

File: Assets/Scripts/Hazard.cs [hunk 13]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+public class Hazard : MonoBehaviour
+{
+
```
Comment (DavidJB526) [inline Assets/Scripts/Hazard.cs]: Remove whitespace

File: Assets/Scripts/Hazard.cs [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+    private void Start()
+    {
+        audioSource = GetComponent<AudioSource>();
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+
+        
+
```
Comment (DavidJB526) [inline Assets/Scripts/Hazard.cs]: Remove whitespace

File: Assets/Scripts/Hazard.cs [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+            Debug.Log("player killed!");
+            // SceneManager.LoadScene(SceneManager.GetActiveScene().name);
+            PlayerMoveScript player = collision.GetComponent<PlayerMoveScript>();
+            player.killed();
+            audioSource.Play();
+        }
+        else
+            Debug.Log("something else entered hazard");
+    }
+
```
Comment (DavidJB526) [inline Assets/Scripts/Hazard.cs]: Remove whitepsace

File: Assets/Scripts/PlayerMoveScript.cs [hunk 16]
```diff
@@ -0,0 +1,276 @@
...
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+
+public class PlayerMoveScript : MonoBehaviour
+{
+    [SerializeField]
+    private Rigidbody2D Thisrigidbody;
+    [SerializeField]
+    private float AccelrationForce =5;
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs:14]: Consider adding ToolTips to clarify floats:
https://docs.unity3d.com/ScriptReference/TooltipAttribute.html

File: Assets/Scripts/PlayerMoveScript.cs [hunk 17]
```diff
@@ -0,0 +1,276 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+
+public class PlayerMoveScript : MonoBehaviour
+{
+    [SerializeField]
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs:11]: Wrap all these SerializeFields up into a region:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-region

File: Assets/Scripts/PlayerMoveScript.cs [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+    }
+    void Flip()
+    {
+        facingRight = !facingRight;
+        Vector3 theScale = transform.localScale;
+        theScale.x *= -1;
+        transform.localScale = theScale;
+
+    }
+
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Remove all unnecessary whitespace from this class

File: Assets/Scripts/pickupGem.cs [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        playercollider = GetComponent<PolygonCollider2D>();
+	}
+	
+	// Update is called once per frame
+	void Update ()
+    {
+		
+	}
+}
```
Comment (DavidJB526) [inline Assets/Scripts/pickupGem.cs:14]: Remove all unnecessary whitespace

**Review by DavidJB526 — CHANGES_REQUESTED** [review]
Looks good so far!  I don't see any issues with the code itself yet, it's mostly just little nitpicky stuff (that I'm only addressing because Antognoli made me do it, too 😄 ).

File: Assets/Scripts/CameraFollowScript.cs (hunk 2 above)
Comment (dantogno) [reply Assets/Scripts/CameraFollowScript.cs:23]: Yes use cinemachine please :)

File: Assets/Scripts/PlayerMoveScript.cs (hunk 16 above)
Comment (dantogno) [reply Assets/Scripts/PlayerMoveScript.cs:14]: https://docs.google.com/document/d/1yrKCY_sVAdWkwgWAjJLiJFj1OOHn6dC__pGkSVB61YY/edit#heading=h.35gp8tzfruuq

Comment (dantogno) [reply Assets/Scripts/PlayerMoveScript.cs:14]: If it's obvious, like initializing the rigid body there, I can see no tool tip. But for values that the designer needs to tweak, tooltips are essential.

Comment (dantogno) [reply Assets/Scripts/PlayerMoveScript.cs:14]: Basically we should have very few if any comments and instead summaries and tool tips.

File: Assets/Scripts/PlayerMoveScript.cs (hunk 17 above)
Comment (dantogno) [reply Assets/Scripts/PlayerMoveScript.cs:11]: https://docs.google.com/document/d/1yrKCY_sVAdWkwgWAjJLiJFj1OOHn6dC__pGkSVB61YY/edit#heading=h.oezlj2pye3ju

**Comment by dantogno** [comment]
Thanks @JieningTao and @DavidJB526! Be sure to let me know when you are done so I can get @JieningTao reviewing someone ASAP!

File: Assets/Scripts/Checkpoint.cs [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+    {
+            if (collision.gameObject.CompareTag("Player") && !isactivated)
+            {
+                Debug.Log("Check Point Reached!");
+                PlayerMoveScript player = collision.GetComponent<PlayerMoveScript>();
+                player.setcurrentcheckpoint(this);
+            
+                audioSource.Play();
+            }
+
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs]: Remove whitespace

File: Assets/Scripts/Checkpoint.cs [hunk 21]
```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Checkpoint : MonoBehaviour
+{
+    [SerializeField]
+    private Color inactiveColor, activeColor;
+    private AudioSource audioSource;
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs:10]: It's generally a good idea to leave a blank line between your SerializeFields and non-SerializeFields, so you might consider adding whitespace here

File: Assets/Scripts/Checkpoint.cs [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+            {
+                Debug.Log("Check Point Reached!");
+                PlayerMoveScript player = collision.GetComponent<PlayerMoveScript>();
+                player.setcurrentcheckpoint(this);
+            
+                audioSource.Play();
+            }
+        
+    }
+    public void Setisactivated(bool value)
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs:42]: Add whitespace

File: Assets/Scripts/Door.cs [hunk 23]
```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class Door : MonoBehaviour
+{
+    [SerializeField]
+    private string sceneToLoad;
+    private bool isPlayerInTrigger;
```
Comment (DavidJB526) [inline Assets/Scripts/Door.cs:11]: Same as before, add space between Serialized and non-Serialized

File: Assets/Scripts/Hazard.cs [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (DavidJB526) [inline Assets/Scripts/Hazard.cs]: Remove whitespace

File: Assets/Scripts/PlayerMoveScript.cs [hunk 25]
```diff
@@ -0,0 +1,253 @@
...
+    // Use this for initialization
+    void Start ()
+    {
+        audioSource = GetComponent<AudioSource>();
+        anim = GetComponent<Animator>();
+        Score = 0;
+    }
+	
+	void Update ()
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Shift line left slightly

File: Assets/Scripts/PlayerMoveScript.cs [hunk 26]
```diff
@@ -0,0 +1,253 @@
...
+    {
+        refilljumps();
+        HandleHorizontalInput();
+        HandleJumpInput();
+
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Remove whitespace

File: Assets/Scripts/PlayerMoveScript.cs [hunk 27]
```diff
@@ -0,0 +1,253 @@
...
+            Updatephysicsmaterial();
+        }
+        else if(Input.GetButtonDown("Activate"))
+        {
+            respawn();
+        }
+        HandleAnimator();
+    }
+
+        void OnTriggerEnter2D(Collider2D other)
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Move line to the left

File: Assets/Scripts/PlayerMoveScript.cs [hunk 28]
```diff
@@ -0,0 +1,253 @@
...
+        {
+            if (!facingRight)
+                return 'R';
+            else
+                return 'L';
+        }
+        return 'N';
+    }
+
+    private void refilljumps()
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Functions should be capitalized

File: Assets/Scripts/PlayerMoveScript.cs [hunk 29]
```diff
@@ -0,0 +1,253 @@
...
+    private void Updatephysicsmaterial()
+    {
+        if (!OnGround() && TouchingWall() != 'N')
+            playergroundcollider.sharedMaterial = playerwallclingPM;
+        else if (Mathf.Abs(HorizontalInput) > 0)
+            playergroundcollider.sharedMaterial = playermovingPM;
+        else
+            playergroundcollider.sharedMaterial = playerstoppingPM;
+
+
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Remove whitespace

File: Assets/Scripts/PlayerMoveScript.cs [hunk 30]
```diff
@@ -0,0 +1,253 @@
...
+    }
+    
+    public void setcurrentcheckpoint(Checkpoint newcurrentcheckpoint)
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Capitalize function

File: Assets/Scripts/PlayerMoveScript.cs [hunk 31]
```diff
@@ -0,0 +1,253 @@
...
+    {
+        if (currentCheckpoint != null)
+            currentCheckpoint.setisactivated(false);
+
+        currentCheckpoint = newcurrentcheckpoint;
+        currentCheckpoint.setisactivated(true);
+    }
+    
+    public void respawn()
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Capitalize function

File: Assets/Scripts/PlayerMoveScript.cs [hunk 32]
```diff
@@ -0,0 +1,253 @@
...
+    {
+        deadText.text = " ";
+        IsDead = false;
+
+     
+
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Remove excess whitespace

File: Assets/Scripts/PlayerMoveScript.cs [hunk 33]
```diff
@@ -0,0 +1,253 @@
...
+        Thisrigidbody.velocity = Vector2.zero;
+
+        if (currentCheckpoint == null)
+            SceneManager.LoadScene(SceneManager.GetActiveScene().name);
+        else
+            transform.position = currentCheckpoint.transform.position;
+
+        //transform.eulerAngles = new Vector3(0, 0, 0);
+        //Thisrigidbody.constraints = RigidbodyConstraints2D.FreezeRotation;
+
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Remove whitespace

File: Assets/Scripts/PlayerMoveScript.cs [hunk 34]
```diff
@@ -0,0 +1,253 @@
...
+    }
+    
+    public void killed()
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Capitalize function

File: Assets/Scripts/PlayerMoveScript.cs [hunk 35]
```diff
@@ -0,0 +1,253 @@
...
+    {
+        deadText.text = "You Died! press E to respawn";
+        IsDead = true;
+
+        Thisrigidbody.constraints = RigidbodyConstraints2D.None;
+
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs:234]: Remove whitespace

File: Assets/Scripts/pickupGem.cs [hunk 36]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class pickupGem : MonoBehaviour
```
Comment (DavidJB526) [inline Assets/Scripts/pickupGem.cs]: Classes should be capitalized, remake this class and remove Functions that aren't used

**Review by DavidJB526 — COMMENTED** [review]
Overall still pretty good!  I don't see anything that jumps out at me, mostly just formatting changes.  Conform.... Conform.... Conform!

File: Assets/Scripts/pickupGem.cs [hunk 37]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PickupGem : MonoBehaviour
+{
+
```
Comment (DavidJB526) [inline Assets/Scripts/pickupGem.cs]: Remove whitespace

File: Assets/Scripts/PlayerMoveScript.cs [hunk 38]
```diff
@@ -0,0 +1,245 @@
...
+        deadText.text = " ";
+        IsDead = false;
+
+        Thisrigidbody.velocity = Vector2.zero;
+
+        if (currentCheckpoint == null)
+            SceneManager.LoadScene(SceneManager.GetActiveScene().name);
+        else
+            transform.position = currentCheckpoint.transform.position;
+
```
Comment (DavidJB526) [inline Assets/Scripts/PlayerMoveScript.cs]: Remove whitespace

File: Assets/Scripts/Checkpoint.cs [hunk 39]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        Color color = inactiveColor;
+        if (isactivated)
+            color = activeColor;
+        spriteRenderer.color = color;
+    }
+    
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+            if (collision.gameObject.CompareTag("Player") && !isactivated)
```
Comment (DavidJB526) [inline Assets/Scripts/Checkpoint.cs]: This whole if statement looks like it should be further to the left

**Review by DavidJB526 — CHANGES_REQUESTED** [review]
Almost done!  Just a last pass over whitespace and shifting over an if statement.  Good work!
